src/js/extras/media-clip-selector/index.js

```diff
@@ -177,19 +175,36 @@ class MediaClipSelector extends MediaChromeHTMLElement {
     this.enableThumbnails();
   }
 
+  connectedCallback() {
+    /** Option 1 */
+    const detail = this.constructor.observedAttributes;
+    const evt = new window.CustomEvent(MediaUIEvents.MEDIA_CHROME_ELEMENT_CONNECTED, { composed: true, bubbles: true, detail });
+    this.dispatchEvent(evt);
+    /** Option 2 */
+    this.setAttribute(MediaUIAttributes.MEDIA_CHROME_ATTRIBUTES, this.constructor.observedAttributes.join(' '));
+  }
+
+  get mediaDuration() {
+    return +this.getAttribute(MediaUIAttributes.MEDIA_DURATION);
```

Comment:
sounds reasonable. This is the only example of a media chrome UI component that is using a property instead of converting the value "on the fly". There were assumptions throughout the code that this will be a (real) number predating this PR. Worth including in the discussion.

Reply:
Yeah, it's interesting, we should probably be opinionated for the sake of consistency, but also now it's really whatever each component needs it to be, so may not really matter.